loopout.go

```diff
@@ -384,19 +393,125 @@ func (s *loopOutSwap) persistState(ctx context.Context) error {
 func (s *loopOutSwap) payInvoices(ctx context.Context) {
 	// Pay the swap invoice.
 	s.log.Infof("Sending swap payment %v", s.SwapInvoice)
-	s.swapPaymentChan = s.lnd.Client.PayInvoice(
+	s.swapPaymentChan = s.payInvoice(
 		ctx, s.SwapInvoice, s.MaxSwapRoutingFee,
 		s.LoopOutContract.UnchargeChannel,
 	)
 
 	// Pay the prepay invoice.
 	s.log.Infof("Sending prepayment %v", s.PrepayInvoice)
-	s.prePaymentChan = s.lnd.Client.PayInvoice(
+	s.prePaymentChan = s.payInvoice(
 		ctx, s.PrepayInvoice, s.MaxPrepayRoutingFee,
 		nil,
 	)
 }
 
+// payInvoice pays a single invoice.
+func (s *loopOutSwap) payInvoice(ctx context.Context, invoice string,
+	maxFee btcutil.Amount,
+	outgoingChannel *uint64) chan lndclient.PaymentResult {
+
+	resultChan := make(chan lndclient.PaymentResult)
+
+	// Extract hash from payment request. Unfortunately the request
+	// components aren't available directly.
+	chainParams := s.lnd.ChainParams
+	hash, _, err := swap.DecodeInvoice(
+		chainParams, invoice,
+	)
+	if err != nil {
+		select {
+		case resultChan <- lndclient.PaymentResult{
```

Comment:
This tries to send to an unbuffered channel that nobody is listening on yet. I don't think this will ever complete. The channel must either be buffered (then the select on `ctx.Done()` can be removed) or the whole send must be done in a subroutine.

Reply:
Good catch, moved into goroutine